lib/datadog/statsd/connection_cfg.rb

```diff
@@ -0,0 +1,64 @@
+module Datadog
+  class Statsd
+    class ConnectionCfg
+      attr_reader :host
+      attr_reader :port
+      attr_reader :socket_path
+      attr_reader :transport_type
+
+      DEFAULT_HOST = '127.0.0.1'
+      DEFAULT_PORT = 8125
+
+      def initialize(host: nil, port: nil, socket_path: nil)
+
+        # Try with constructor args
+        if setup_with(host, port, socket_path)
+          return
+        end
+
+        # Try with env vars
+        if setup_with(
+            ENV['DD_AGENT_HOST'],
+            ENV['DD_DOGSTATSD_PORT'].nil? ? nil : ENV['DD_DOGSTATSD_PORT'].to_i,
+            ENV['DD_DOGSTATSD_SOCKET'])
+          return
+        end
+
+        # Fall back to defaults
+        setup_with(DEFAULT_HOST, DEFAULT_PORT, nil)
+      end
```

Comment:
Minor: I usually advocate for trying to turn comments such as the ones in this method into actual code (of course, this does not apply to all comments -- but I think it applies to the comments here).

For instance, what do you think of the following alternative implementation of this method?

```ruby
def initialize(host:, port:, socket_path:)
  try_with_contructor_args(host, port, socket_path) || try_with_env_vars || use_defaults
end
```

Reply:
Oh, I like that!  I am just starting to read [Five Lines of Code](https://www.manning.com/books/five-lines-of-code) and this is right out of the first chapter :)